tests/test_utils.py

```diff
@@ -169,6 +171,21 @@ def test_get_ssl_cert_file():
     assert timer.elapsed.total_seconds() == pytest.approx(0.1, abs=0.05)
 
 
+@pytest.mark.parametrize(
+    ["no_proxy", "exempt_list"],
+    [
+        ({}, []),
+        ({"NO_PROXY": ""}, []),
+        ({"NO_PROXY": "http://127.0.0.1"}, ["http://127.0.0.1"]),
```

Comment:
I've been searching for a formal definition of `NO_PROXY` and [the closest I could find](https://curl.haxx.se/docs/manual.html) suggests:

> A comma-separated list of host names that shouldn't go through any proxy is set in (only an asterisk, * matches all hosts)

So presumably the `https://` prefix should be invalid if we abide for that rule?

Reply:
added a test that shows that the https:// prefix is invalid
no_proxy is not retrieved via urllibs getproxies() function